pymc3/distributions/simulator.py

```diff
@@ -51,16 +101,33 @@ def random(self, point=None, size=None):
         -------
         array
         """
+        params = draw_values([*self.params], point=point, size=size)
+        if size is None:
+            return self.function(*params)
+        else:
+            return np.array([self.function(*params) for _ in range(size)])
+
+    def gaussian_kernel(self, obs_data, sim_data):
+        return np.sum(-0.5 * ((obs_data - sim_data) / self.epsilon) ** 2)
 
-        raise NotImplementedError("Not implemented yet")
+    # we are assuming obs_data and sim_data are already sorted!
+    def wasserstein(self, obs_data, sim_data):
+        return np.mean(np.abs((obs_data - sim_data) / self.epsilon))
+
+    # we are assuming obs_data and sim_data are already sorted!
+    def energy(self, obs_data, sim_data):
+        return 1.4142 * np.mean(((obs_data - sim_data) / self.epsilon) ** 2) ** 0.5
```

Comment:
recommend isolating it into separate functions instead of methods of a single class